plugins/examples/custom-settings/build.gradle

```diff
@@ -17,15 +17,21 @@
  * under the License.
  */
 
-apply plugin: 'elasticsearch.esplugin'
+plugins {
+  id 'elasticsearch.esplugin'
```

Comment:
I still don't like having this using the plugins dsl until we actually have build-tools published to the gradle plugin portal. While these examples currently are missing the buildtools section to make apply work outside the build, if they are copied with the plugins block they will never work, and the purpose of these is to make the examples copyable.

Reply:
I'm not entirely happy about it either, but it handles the injection of the plugin that was just built, a lot of boilerplate that this DSL and testKit helps us avoid. That being said, I think with what we have now, we could inject the local repo into the build script classpath and easily avoid that boilerplate. I will take a look.